packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -1113,8 +1113,11 @@ interface StackDependency {
 }
 
 function extractSingleValue<T>(array: T[] | undefined): T[] | T | undefined {
-  if (array && array.length === 1) {
-    return array[0];
+  if (array) {
+    if (array.length === 1) {
+      return array[0];
+    }
+    return Array.from(new Set(array)); // Unique values
```

Comment:
Something about this function is very odd. It says "extract single value" but can return an array. Also, there is no mention of the deduplication.

Maybe we should just remove this function and write this very purpose-built code when synthing the Transform

Reply:
Agree. Also how about having a `addTransform()` method on `Stack`? Since `templateOptions.transforms` is writable and is a list since #3395 this would avoid having to do `this.stack.templateOptions.transforms = [...this.stack.templateOptions.transforms || [], 'ExtraTransform']` each time...

Reply:
Yes!

Reply:
done already